### PR TITLE
Use the skeleton from the NPC's set model (bug #4747)

### PR DESCRIPTION
**Pull request by Capostrophic**
[Bug 4747](https://gitlab.com/OpenMW/openmw/issues/4747).

Previously only the animations and not the skeleton were used from the mModel field in NPC record. This caused NPCs with changed model often lack the nodes necessary to perform the animations from the models or use set body parts of their race.

Now non-werewolf NPCs will use the skeleton of the model set in their record and fallback to their default skeleton otherwise; animations of the default skeleton will be added too, "just in case". As I said under the report, rot has confirmed this should be fine.

This is Teleri Helvi with Khajiit race with Beast flag unchecked. rot's files allow NPCs with it assigned to have tail if it's assigned to their race.
screenshot024

In master, furry Teleri lacks a tail and there are multiple errors and warnings logged:
```
Error: RigGeometry did not find bone Bip01 L Toe0                                                                                                                                                                                            
Error adding NPC part: Can't find attachment node Tail                                                                                                                                                                                       
Warning: addAnimSource: can't find bone 'bip01 l toe0' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)
Warning: addAnimSource: can't find bone 'bip01 r toe0' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)                                                                                                    
Warning: addAnimSource: can't find bone 'bip01 tail' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)                                                                                                      
Warning: addAnimSource: can't find bone 'bip01 tail1' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)                                                                                                     
Warning: addAnimSource: can't find bone 'bip01 tail2' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)                                                                                                     
Warning: addAnimSource: can't find bone 'bip01 tail3' in meshes\xbase_anim_female.nif (referenced by meshes\xbase_animkna_importable.kf)       
```
With these changes they are no more.

### CODE REVIEW

**Comment by Capostrophic** [comment]
Resolved the merge conflict. Is there anything that I missed or needs attention?

**Comment by Capostrophic** [comment]
It appears that Animated Morrowind and Arktwend need this too - there are a bunch of bone-related warnings for Animated Morrowind and Arktwend animations when running Arktwend about standard skeletons lacking tail and left/right toe bones, although they don't seem to use them unlike rot's replacement animations.

File: apps/openmw/mwrender/npcanimation.cpp [hunk 1]
```diff
@@ -464,8 +464,12 @@ void NpcAnimation::updateNpcBase()
     bool is1stPerson = mViewMode == VM_FirstPerson;
     bool isBeast = (race->mData.mFlags & ESM::Race::Beast) != 0;
 
-    std::string smodel = SceneUtil::getActorSkeleton(is1stPerson, isFemale, isBeast, isWerewolf);
-    smodel = Misc::ResourceHelpers::correctActorModelPath(smodel, mResourceSystem->getVFS());
+    std::string defaultSkeleton = SceneUtil::getActorSkeleton(is1stPerson, isFemale, isBeast, isWerewolf);
+    defaultSkeleton = Misc::ResourceHelpers::correctActorModelPath(defaultSkeleton, mResourceSystem->getVFS());
```
Comment (terabyte25) [inline apps/openmw/mwrender/npcanimation.cpp]: Why not assign to sModel instead of defaultSkeleton?

File: apps/openmw/mwrender/npcanimation.cpp [hunk 2]
```diff
@@ -481,15 +485,13 @@ void NpcAnimation::updateNpcBase()
...
 
-        if(!isWerewolf)
-        {
-            if(mNpc->mModel.length() > 0)
-                addAnimSource(Misc::ResourceHelpers::correctActorModelPath("meshes\\" + mNpc->mModel, mResourceSystem->getVFS()), smodel);
-            if(Misc::StringUtils::lowerCase(mNpc->mRace).find("argonian") != std::string::npos)
-                addAnimSource("meshes\\xargonian_swimkna.nif", smodel);
-        }
+        if(!isWerewolf && Misc::StringUtils::lowerCase(mNpc->mRace).find("argonian") != std::string::npos)
+            addAnimSource("meshes\\xargonian_swimkna.nif", smodel);
```
Comment (psi29a) [inline apps/openmw/mwrender/npcanimation.cpp]: ugh... hardcoded stuff... we should get away from this ASAP

**Pull request merged**